### DISCUSSION TITLE
Assistance with Building VSCodium for RISC-V

### DISCUSSION THREAD

**Discussion by karthick-govindaraj** [VSCodium/vscodium#2098] [General]
I am currently working with RISC-V and successfully running VSCodium on my RISC-V board. However, I'm facing challenges with building VSCodium from source on both my RISC-V board and a Docker QEMU emulator (https://github.com/DavidBurela/riscv-emulator-docker-image.git), where I encounter a yarn error.

I followed the build instructions here: https://github.com/VSCodium/vscodium/blob/master/docs/howto-build.md, but I haven’t been able to resolve the error. Would you be able to assist and provide guidance on how you managed to build VSCodium for RISC-V?

Thank you very much for your time and any insights you can offer.

**Comment by daiyam** [comment]
What's the error?

**Reply by karthick-govindaraj** [reply]
![image](https://github.com/user-attachments/assets/f18c3d58-5988-4bb6-9a5d-00f0363ae5d1)

I didn't get an error message, but the compilation failed and the process was killed.

**Reply by daiyam** [reply]
I see only

> the process was killed

**Reply by karthick-govindaraj** [reply]
@daiyam ,I have updated the machine and tried to rebuild the application. Now, I've encountered the following error:

![Screenshot from 2024-11-12 12-00-12](https://github.com/user-attachments/assets/18223579-939d-40b5-bd7e-ddb85523cb4f)

**Reply by karthick-govindaraj** [reply]
hi @daiyam,
could you please check above issue while building vscodium and help resolve the above issue? It seems that a file or dependency required by Electron isn’t available at the specified URL. I’ve rebuilt it twice but am still stuck at this point. Could you guide me on how to fix this error?

**Reply by daiyam** [reply]
try with `export VSCODE_ELECTRON_REPOSITORY='riscv-forks/electron-riscv-releases'` to download the correct electron.

**Reply by karthick-govindaraj** [reply]
Thanks, @daiyam, that worked. However, I'm now encountering another error: HttpError: request to https://api.github.com/repos/riscv-forks/electron-riscv-releases/releases/180458845/assets failed, reason: socket hang up

![image](https://github.com/user-attachments/assets/eaf58291-ad4a-4c38-8d6a-ec672c670c1a)

but the link works fine I don't know whats gone wrong

![image](https://github.com/user-attachments/assets/50b479ff-2a0d-4f39-a838-e705d33b5c37)

**Reply by karthick-govindaraj** [reply]
@daiyam  After resolving several errors, I am now stuck with this issue. Could you please guide me on how to fix it?

![image](https://github.com/user-attachments/assets/774c0a46-6fed-4222-a428-bccf0d5d7c4c)

**Reply by karthick-govindaraj** [reply]
Hi @daiyam, do you have any idea about the above error?
